src/wilson/dom.cljs

```diff
@@ -78,6 +79,50 @@
   ([ks rows]
    (table ks rows {})))
 
+
+(defn sorted-table
+  "Returns a sortable table (using `wilson.dom/table`) with rows sorted using
+  `wilson.dom/sort-rows`. Table headers will have an on-click handler that
+  will change the sorting key and/or order (those values are saved under
+  unique names in the passed state), as well as a class of `asc` or `desc`,
+  when table is ordered by corresponding key (refer to
+  `resources/public/site.css` for example table styling). You can pass the
+  same options  map that `wilson.dom/table` would accept. In addition, you can
+  extend options map with `:sort-fns` - its value will get passed directly to
+  `wilson.dom/sort-rows."
+  ([ks rows state opts]
+   (let [sort-key-id (gensym "wilson-order-key")
+         sort-order-id (gensym "wilson-order-key")]
+     (swap! state merge {sort-key-id (first ks)
+                         sort-order-id :asc})
+     (fn []
+       (let [state-deref @state
+             sort-fns (or (:sort-fns opts)
+                       {:default (fn [k rows]
+                                  (if (= (sort-order-id state-deref) :asc)
+                                      (sort-by k rows)
+                                      (reverse (sort-by k rows))))})
+             sorted-rows (sort-rows rows sort-fns (sort-key-id state-deref))
+             get-new-order
+              (fn [state k]
+               (let [state-deref @state
+                     swap-order {:asc :desc :desc :asc}]
+                {sort-key-id k
+                 sort-order-id (if (= (sort-key-id state-deref) k)
+                                    (swap-order (sort-order-id state-deref))
```

Comment:
Good point. Tho, I'm not sure I see value in moving it to a top-level function. It will need to know about ids generated with gensym and that will make the swap call a little longer (at the same time, I don't think we are likely to use this fn anywhere else). Let me know what you think about my changes